controllers/commonservice_controller.go

```diff
@@ -209,6 +233,50 @@ func checkKeyBeforeMerging(key string, defaultMap interface{}, changedMap interf
 	}
 }
 
+func (r *CommonServiceReconciler) finalizeCommonService() error {
+	klog.Info("Uninstalling common services")
+
+	// Remove common service OperandRequest
+	opreq := common.NewUnstructured("operator.ibm.com", "OperandRequest", "v1alpha1")
+	if err := r.Client.DeleteAllOf(ctx, opreq, client.InNamespace("ibm-common-services"), client.MatchingLabels{"ibm-common-services.common-service/registry": "true"}); err != nil && !errors.IsNotFound(err) {
+		klog.Error("Delete OperandRequest failed: ", err)
+		return err
+	}
+	// Remove common service OperandConfig
+	opcon := common.NewUnstructured("operator.ibm.com", "OperandConfig", "v1alpha1")
+	opcon.SetName("common-service")
+	opcon.SetNamespace("ibm-common-services")
+	if err := r.Client.Delete(ctx, opcon); err != nil && !errors.IsNotFound(err) {
+		klog.Error("Delete OperandConfig failed: ", err)
+		return err
+	}
+	// Remove common service OperandRegistry
+	opreg := common.NewUnstructured("operator.ibm.com", "OperandRegistry", "v1alpha1")
+	opreg.SetName("common-service")
+	opreg.SetNamespace("ibm-common-services")
+	if err := r.Client.Delete(ctx, opreg); err != nil && !errors.IsNotFound(err) {
+		klog.Error("Delete OperandRegistry failed: ", err)
+		return err
+	}
+
+	// Remove ODLM sub and csv
+	if err := common.DeleteExistingODLM(r.Reader, r.Client); err != nil && !errors.IsNotFound(err) {
```

Comment:
`DeleteExistingODLM` seems only delete the odlm in the ibm-common-service namespace. Should we delete it from the `openshift-operators` namespace as well?

Reply:
@horis233 For the which resources need to be deleted, I think we need some discuss with @chenzhiwei.
This PR only provide a cleanup framework and UT, after this PR merged, we can add more detail cleanup func into func `finalizeCommonService()`